packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -334,17 +332,20 @@ function setElementShadowToken(elm: Element, token: string) {
 }
 
 // Set the scope token class for *.scoped.css styles
-function setScopeTokenClassIfNecessary(elm: Element, owner: VM) {
+function setScopeTokenClassIfNecessary(elm: Element, owner: VM, renderer: RendererAPI) {
     const { cmpTemplate, context } = owner;
+    const { getClassList } = renderer;
     const token = cmpTemplate?.stylesheetToken;
     if (!isUndefined(token) && context.hasScopedStyles) {
+        // TODO [#2762]: this dot notation with add is probably problematic
```

Comment:
You are right, the token comes from user land. The user can tinker with the compiled template and change the token. So it must be run through the hooks.